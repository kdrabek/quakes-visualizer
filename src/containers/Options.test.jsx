import React from 'react';
import Immutable from 'immutable';
import { shallow } from 'enzyme';

import {
  OptionsContainer, mapStateToProps, mapDispatchToProps,
} from './Options';

describe('(Container) Options', () => {
  const state = Immutable.Map();

  const initialProps = mapStateToProps(state);
  const initialActions = mapDispatchToProps(p => p);
  const props = { ...initialActions, ...initialProps };

  it('has right props', () => {
    expect(initialProps).toMatchSnapshot();
  });


  it('has right actions', () => {
    expect(initialActions).toMatchSnapshot();
  });

  it('should render', () => {
    const wrapper = shallow(<OptionsContainer {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
