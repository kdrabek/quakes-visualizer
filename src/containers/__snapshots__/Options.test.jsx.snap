// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Container) Options has right actions 1`] = `
Object {
  "actions": Object {
    "getEarthquakes": [Function],
    "updateMap": [Function],
  },
}
`;

exports[`(Container) Options has right props 1`] = `Object {}`;

exports[`(Container) Options should render 1`] = `
<withState(withState(withHandlers(Options)))
  handleOnClick={[Function]}
/>
`;
